pybop/models/base_model.py

```diff
@@ -204,15 +206,16 @@ def simulateS1(self, inputs, t_eval):
                 calculate_sensitivities=True,
             )
 
-            return (
-                sol[self.signal].data,
-                np.asarray(
-                    [
-                        sol[self.signal].sensitivities[key].toarray()
-                        for key in self.fit_keys
-                    ]
-                ).T,
-            )
+            predictions = [sol[signal].data for signal in self.signal]
+
+            sensitivities = [
+                np.array(
+                    [[sol[signal].sensitivities[key]] for signal in self.signal]
+                ).reshape(len(sol[self.signal[0]].data), self.n_outputs)
```

Comment:
Do we have an easier way to get `len(sol[self.signal[0].data])`? Maybe it can be taken from `self.target`? Looks good if not, just wondering if it could be simplified.

Reply:
Yeah, I thought the same but ended up with this expression to ensure that it still works if the simulation terminates early. In this case, I think the length has to be taken from the newly-simulated `sol`.